R/get_rcode_utils.R

```diff
@@ -4,21 +4,33 @@
 #'
 #' @return Character object contain code
 #' @keywords internal
-get_rcode_libraries <- function() {
-  vapply(
-    utils::sessionInfo()$otherPkgs,
-    function(x) {
-      paste0("library(", x$Package, ")")
-    },
-    character(1)
-  ) %>%
-    # put it into reverse order to correctly simulate executed code
-    rev() %>%
-    paste0(sep = "\n") %>%
-    paste0(collapse = "")
-}
+get_rcode_libraries <- function(dataset_rcode) {
+  packages <- rev(vapply(utils::sessionInfo()$otherPkgs, base::`[[`, character(1), "Package"))
+
+  parsed_libraries <- c()
+  if (!missing(dataset_rcode)) {
+    # Extract all lines with library()
+    #  TODO: remove strings first as this will pass "this is a string with library(something) in it"
+    user_libraries <- Filter(
+      function(.x) grepl("library\\(.*\\)$", .x),
+      vapply(strsplit(dataset_rcode, "\n")[[1]], trimws, character(1))
+    )
 
+    # Keep only library name
+    parsed_libraries <- gsub(
+      # library(...) must be located at beginning of line, or have a valid character before
+      "(^l|.*<-|.*[ ;=\\({]l)ibrary\\(([a-z][a-zA-Z0-9.]*)\\)$", "\\2",
```

Comment:
Can `[a-z][a-zA-Z0-9.]` be replaced with `[:alnum:]`?

Reply:
Yes, `[[:alnum:].]` due to dot

Reply:
Right, I missed the full stop when retyping :+1:

Reply:
Wait, I think I misunderstood your reply to my comment. What I meant to ask is whether you can replace 
```
"(^l|.*<-|.*[ ;=\\({]l)ibrary\\(([a-z][a-zA-Z0-9.]*)\\)$"
```
with 
```
"(^l|.*<-|.*[ ;=\\({]l)ibrary\\(([:alnum:]*)\\)$"
```

I don't quite get why you use `<any lowercase letter><any number of letters/digits/anychar>`.

Reply:
But I see now that things like `[:alnum:]` are not entirely reliable.

Reply:
I misunderstood the documentation about the first letter.

A valid package name needs to be `<letter><any number, letter and dot>`